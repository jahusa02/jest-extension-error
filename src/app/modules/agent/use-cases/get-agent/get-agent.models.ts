export interface RequestModel {
  user: {
    id: string;
  };
}

export interface ResponseModel {
  agent: {
    id: string;
    firstName: string;
    lastName: string;
    locale: string;
  };
}
