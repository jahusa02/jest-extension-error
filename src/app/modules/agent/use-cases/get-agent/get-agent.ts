/* eslint-disable @typescript-eslint/require-await */
import { RequestModel, ResponseModel } from './get-agent.models';
import { AgentEntity } from '../../../../entities/agent/agent';

export class GetAgentUseCase {
  constructor() {}

  async execute(_requestModel: RequestModel): Promise<ResponseModel> | never {
    const result: AgentEntity = null as unknown as AgentEntity;

    if (result) {
      throw result;
    }

    return {
      agent: {
        id: '',
        firstName: '',
        lastName: '',
        locale: '',
      },
    };
  }
}
