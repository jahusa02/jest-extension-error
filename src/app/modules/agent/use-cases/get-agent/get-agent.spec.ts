import { GetAgentUseCase } from './get-agent';
import { RequestModel } from './get-agent.models';

describe('use-case get agent', () => {
  let useCase: GetAgentUseCase;

  let userId: string;
  let requestModel: RequestModel;

  beforeEach(() => {
    userId = 'MyUserId';
    requestModel = {
      user: {
        id: userId,
      },
    };
  });

  it.skip('should be created', () => {
    expect(useCase).toBeInstanceOf(GetAgentUseCase);
  });

  describe('execute', () => {
    it.skip('should call entities from the gateway', async () => {
      await useCase.execute(requestModel);
    });

    describe('gateway returns an agent entity', () => {
      it('should return responseModel with agent when gateway returns an entity', async () => {
        expect(null).toMatchSnapshot();
      });
    });

    describe('gateway returns a failure', () => {
      it('should throw an error if gateway returns a failure', () => {
        expect(null).toBe(null);
      });
    });
  });
});
