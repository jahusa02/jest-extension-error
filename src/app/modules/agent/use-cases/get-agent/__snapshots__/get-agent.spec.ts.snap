// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`use-case get agent execute gateway returns an agent entity should return responseModel with agent when gateway returns an entity 1`] = `null`;
